SharpBucket/SharpBucket.cs

```diff
@@ -119,24 +83,24 @@ private T Send<T>(T body, Method method, string overrideUrl = null, IDictionary<
             return response;
         }
 
-        internal T Get<T>(T body, string overrideUrl, object requestParameters = null)
+        public T Get<T>(T body, string overrideUrl, object requestParameters = null)
```

Comment:
Do we really what to expose that methods to public ?

If for unit test we want to mock them we may declare them in an internal interface instead of changing their accessibility type

```suggestion
internal interface ISharpBucketInternal : ISharpBucket
{
    T Get<T>(T body, string overrideUrl, object requestParameters = null);
}

public class SharpBucket : ISharpBucketInternal
{
 ...
}

```

Reply:
I've checkout that branch to have a closer look at it, and now I understand why tath methods have been publicly expose, since they are needed by all EndPoint classes in which we should inject the public interface

However the signature of that methods are not really good and had force to create empty instances for the `T body` parameter in a large portion of the code, so I prefer to rework that signatures before going in the direction of publicly expose anything to avoid future breaking changes

So I will let the interfacing of Sharpbucket on standby and improve other things before